src/components/app/details/cicdHistory/Sidebar.tsx

```diff
@@ -106,16 +111,17 @@ const HistorySummaryCard = React.memo(
         type,
         stage,
     }: HistorySummaryCardType): JSX.Element => {
-        const { path, url } = useRouteMatch()
+        const match = useRouteMatch()
         const { pathname } = useLocation()
         const currentTab = pathname.split('/').pop()
-        const { triggerId, ...rest } = useParams<{ triggerId: string }>()
+        const { triggerId, envId, ...rest } = useParams<{ triggerId: string, envId: string }>()
 
         const getPath = (): string => {
             if (type === HistoryComponentType.CD) {
-                return generatePath(path, { ...rest, triggerId: id }) + '/' + currentTab
-            } else {
-                return generatePath(path, { ...rest, buildId: id }) + '/' + currentTab
+                return generatePath(match.path, { ...rest, triggerId: id }) + '/' + currentTab
```

Comment:
should we not use if else for payload generation rather than whole path generation